fre/pp/configure_script_xml.py

```diff
@@ -174,67 +193,93 @@ def main(xml, platform, target, experiment, do_analysis, historydir, refinedir,
-                                                                        platform,
-                                                                        target,
-                                                                        experiment)
-    logging.info(">> {}".format(fetch_pp_cmd))
-    fetch_pp_process = subprocess.run(fetch_pp_cmd,
-                                      shell=True,
-                                      check=True,
-                                      capture_output=True,
-                                      universal_newlines=True)
-    ppDir = fetch_pp_process.stdout.strip()
-    logging.info(ppDir)
-
-    fetch_analysis_dir_cmd = "frelist -x {} -p {} -t {} {} -d analysis".format(xml,
-                                                                        platform,
-                                                                        target,
-                                                                        experiment)
-    logging.info(">> {}".format(fetch_analysis_dir_cmd))
-    fetch_analysis_dir_process = subprocess.run(fetch_analysis_dir_cmd,
-                                      shell=True,
-                                      check=True,
-                                      capture_output=True,
-                                      universal_newlines=True)
-    analysisDir = fetch_analysis_dir_process.stdout.strip()
-    logging.info(analysisDir)
+    if historydir is None:
+        logging.info("Running frelist for historydir assignment, this may fail")
+        logging.info("If so, try the frelist call manually and use the historydir argument")
+
+        fetch_history_cmd = f"frelist -x {xml} -p {platform} -t {target} {experiment} -d archive"
+        logging.info(">> %s", fetch_history_cmd)
+
+        fetch_history_process = subprocess.run(fetch_history_cmd,
+                                               shell=True,
+                                               check=True,
+                                               capture_output=True,
+                                               universal_newlines=True)
+        historydir = fetch_history_process.stdout.strip() + '/history'
+        logging.info(historydir)
 
-    gridSpec = frelist_xpath(xml, platform, target, experiment, do_analysis, historydir, refinedir, ppdir, do_refinediag, pp_start, pp_stop, validate, verbose, quiet, dual, 'input/dataFile[@label="gridSpec"]')
-    simTime = frelist_xpath(xml, platform, target, experiment, do_analysis, historydir, refinedir, ppdir, do_refinediag, pp_start, pp_stop, validate, verbose, quiet, dual, 'runtime/production/@simTime')
-    simUnits = frelist_xpath(xml, platform, target, experiment, do_analysis, historydir, refinedir, ppdir, do_refinediag, pp_start, pp_stop, validate, verbose, quiet, dual, 'runtime/production/@units')
+
+
+    # Q: should respond to do_refineDiag and refinedir args of this function?
```

Comment:
It's not an answer, but in general we want to hide all refinediag stuff and hope no one misses it. Preserve the capability and discourage its use. :)

Reply:
reasonable!